### PR TITLE
Structure Recipes

### PR DESCRIPTION
**Pull request by Tinfy-Gimpassa**
Full change log:

Leather Couch Blue, Couch Panels Grey, Chair Large Red/Grey, Chaise Lounge Orange, Bench Cushions Leather Brown, Couch Cushions Grey/Red, Couch Blanket Covered Grey/Red, all made sittable on use using zep sit script. erf attached.

Chancellor Chair adjusted to be sittable. resref in crafting recipe adjusted. Erf attached.

Furniture Category Type "Beds" was not populating in-game. Found the fix in Recipe Category Type and fixed.

Added "Walls" as a furniture type in Recipe Category Type.

Added the following:
Rug Classic Light Brown
Pile of Cushions White
Oven 
Coffee Maker
Microwave Black
Bar Wooden
Couch Leather Blue
Bench Leather Cushion Brown
Table Plastic Large
Table Round Oak
Metal Wall Wide Light
Store Counter Stained
Chair Large Grey Red
Bunkbed Metal Grey
Shower White
Shower Floor Basin
Couch Leather Panels Grey
Shelves Warehouse Full
Bookshelf Jedi
Metal Wall Single Pipes
Wooden Wall Planks Small
Television Big Screen
Television Old Model
Foyer Chandelier
Metal Wall Door Light
Metal Wall Wide Panels
Metal Wall Door Panels
Metal Wall Single Light
Metal Wall Single Ribbed
Metal Wall Wide Ribbed
Metal Wall Door Ribbed
Metal Wall Wide Pipes
Metal Wall Door Pipes
Couch Cushion Grey Red
Bench Elegant Grey
Chaise Lounge Orange
Couch Blanket Cover Red
Chair Plinth
Chair Dining Grey,
Chaise Lounge Red
Chair Dining Orange
Table Polygon Design
Table Round Glass
Table Dark Glass

### CODE REVIEW

**Comment by zunath** [comment]
Compilation errors on this PR. Contributor has been notified in Discord.

File: SWLOR.Game.Server/Service/CraftService/RecipeCategoryType.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public enum RecipeCategoryType
         Rifle = 23,
         [RecipeCategory("Twin Blade", true)]
         TwinBlade = 24,
-        [RecipeCategory("Bed", true)]
+        [RecipeCategory("Beds", true)]
```
Comment (zunath) [inline SWLOR.Game.Server/Service/CraftService/RecipeCategoryType.cs:57]: This is really minor but all of the recipe categories are singular for consistency sake. Can you swap this back to "Bed" and rename "Walls" to "Wall" below?

File: SWLOR.Game.Server/Service/CraftService/RecipeType.cs [hunk 2]
```diff
@@ -607,6 +607,53 @@ public enum RecipeType
         FabricationTerminal = 1203,
         Refinery = 1204,
         SmitheryBench = 1205,
+        RugClassicLightBrown = 1206,
+        PileofCushionsWhite = 1207,
+        Oven = 1208,
```
Comment (zunath) [inline SWLOR.Game.Server/Service/CraftService/RecipeType.cs:612]: You have two entries for 1208 here which throw off all subsequent numbers.

File: SWLOR.Game.Server/Service/CraftService/RecipeType.cs [hunk 3]
```diff
@@ -607,6 +607,53 @@ public enum RecipeType
...
+        ChaiseLoungeOrange = 1240,
+        CouchBlanketCoverRed = 1241,
+        ChairPlinth = 1242,
+        ChairDiningGrey = 1243,
+        ChaiseLoungeRed = 1244,
+        ChairDiningOrange = 1245,
+        TablePolygonDesign = 1246,
+        TableRoundGlass = 1247,
+        TableDarkGlass = 1248,
+
```
Comment (zunath) [inline SWLOR.Game.Server/Service/CraftService/RecipeType.cs:654]: nitpick: You have a bunch of extra empty lines here that can be deleted.

File: SWLOR.Game.Server/Service/PropertyService/StructureType.cs [hunk 4]
```diff
@@ -1272,10 +1272,280 @@ public enum StructureType
...
+            "",
+            true,
+            1)]
+        WoodenWallPlanksSmall = 245,
+        [Structure("Television, Big Screen",
+            "_mdrn_pl_tvbgscr",
+            "",
+            true,
+            1)]
+        TelevisionBigScreen = 245,
```
Comment (zunath) [inline SWLOR.Game.Server/Service/PropertyService/StructureType.cs:1407]: Duplicate 245 here which throws off the rest of the numbers

File: SWLOR.Game.Server/Service/PropertyService/StructureType.cs [hunk 5]
```diff
@@ -1272,10 +1272,280 @@ public enum StructureType
...
+            "",
+            true,
+            1)]
+        CouchBlanketCoverRed = 260,
+        [Structure("Chair, Plinth",
+            "_mdrn_pl_chairel",
+            "",
+            true,
+            1)]
+        ChairPlinth = 260,
```
Comment (zunath) [inline SWLOR.Game.Server/Service/PropertyService/StructureType.cs:1503]: Duplicate 260 here

**Review by zunath — CHANGES_REQUESTED** [review]
Made a few comments on things that need to be fixed. Also, I reformatted the FurnitureRecipes.cs file because evidently the code wasn't lined up properly - I think this was there before your changes.

There's a couple more things you need to do for this:

1.) Simply sending up the .erf files doesn't help me. I ask people to do this if they aren't going to the trouble of submitting a PR. But since you're doing a PR I need you to import these to the module and submit the changes. The process is:

- Ensure you have the latest module 
- Open the module up in the toolset
- Import your erf files
- Save the module from the toolset
- Copy the mod file to the Module folder of the repository
- Run the UnpackModule.cmd script
- It will generate JSON files which you commit and push into your branch/PR
Full details are here: https://wiki.starwarsnwn.com/Development/packing-and-unpacking-module

2.) The new furniture items have to be added to the module as well. Take a look at the Furniture category under items to see how these are made. You can probably get away with right clicking an existing one and selecting "Edit Copy". Then change the Name, Tag, and Resref. These need to be included in the unpack process mentioned above.

File: SWLOR.Game.Server/Service/CraftService/RecipeCategoryType.cs (hunk 1 above)
Comment (Tinfy-Gimpassa) [reply SWLOR.Game.Server/Service/CraftService/RecipeCategoryType.cs:57]: note to self: done

File: SWLOR.Game.Server/Service/PropertyService/StructureType.cs (hunk 5 above)
Comment (Tinfy-Gimpassa) [reply SWLOR.Game.Server/Service/PropertyService/StructureType.cs:1503]: note to self: corrected

File: SWLOR.Game.Server/Service/PropertyService/StructureType.cs (hunk 4 above)
Comment (Tinfy-Gimpassa) [reply SWLOR.Game.Server/Service/PropertyService/StructureType.cs:1407]: note to self: remenant from replacing wooden wall planks (small) with metal wall. reason: metal walls made more sense then wooden. wooden wall recipe fully cleared and deleted from all 3 files.

File: SWLOR.Game.Server/Service/CraftService/RecipeType.cs (hunk 3 above)
Comment (Tinfy-Gimpassa) [reply SWLOR.Game.Server/Service/CraftService/RecipeType.cs:654]: note to self: corrected. followed formatting layout to put // Structures: 1496-1516 above following segment for structure category

File: SWLOR.Game.Server/Service/CraftService/RecipeType.cs (hunk 2 above)
Comment (Tinfy-Gimpassa) [reply SWLOR.Game.Server/Service/CraftService/RecipeType.cs:612]: note to self: corrected,  1226 left free as result of deleting wooden wall planks. will put something here later.

File: SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs [hunk 6]
```diff
@@ -297,1943 +297,2434 @@ private void Tier1()
...
+                .Resref("structure_0227")
+                .Level(21)
+                .Quantity(1)
+                .RequirementPerk(PerkType.FurnitureBlueprints, 2)
+                .EnhancementSlots(RecipeEnhancementType.Structure, 2)
+                .Component("fiberpl_flawed", 2)
+                .Component("elec_flawed", 3);
+
+            // Television, Old Model
+            _builder.Create(RecipeType.CoffeeMaker, SkillType.Fabrication)
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs:1589]: This is the wrong enumeration for the Television, Old Model. Please correct it with the right enum.

**Comment by zunath** [comment]
I've spent a few hours cleaning this up today. There were a number of issues I had to resolve.

- Recipes were out of order in the FurnitureRecipes.cs class. These should be ordered by level under the appropriate tier.
- Several recipes were missing enhancement slots
- Several recipes had incorrect resrefs
- Several recipes had incorrect components for their level/tier
- The erfs were never imported so the placeables you added never actually got added to this PR.
- Several recipes had a level higher than their tier. A few were in the 20s which should be tier 3 but they were marked as tier 2.
- The structure items were not included in the erf files so I had to add these manually.
- These new recipe lines were not added to the design bible.
- One of the recipes is missing a placeable. I had to remove this for now because I have no idea what appearance you intended.

I mention these not to complain - but to give you a head's up in the future if you decide to do another one of these. I would suggest filling out the new lines in the bible spreadsheet first, then using our tool to convert that info into code. It's a lot easier than doing these by hand now.

I'm wrapping up the fixes now and will merge with the intention of releasing this in the next update.

**Review by zunath — APPROVED** [review]

**Pull request merged**